estuary/utils/story.py

```diff
@@ -144,6 +145,51 @@ def get_sibling_nodes_count(self, results, reverse=False):
             return correlated_nodes[::-1]
         return correlated_nodes
 
+    def get_wait_times(self, results):
+        """
+        Get the wait time between two different artifacts.
+
+        :param list results: contains inflated results from Neo4j
+        :return: list of wait time tuples in order of the pipeline (oldest to newest)
+        :rtype: list
+        """
+        len_story = len(results)
+        if len_story < 2:
+            raise RuntimeError('This function can\'t be called with one or zero elements')
+
+        # Some services do not have a real completion time because they perform a single action
+        # that takes a negligible amount of time
+        completion_times = {
+            'BugzillaBug': 'creation_time',
+            'DistGitCommit': 'commit_date',
+            'Advisory': 'created_at',
+            'ContainerAdvisory': 'created_at',
+            'FreshmakerEvent': 'time_done',
+            'KojiBuild': 'completion_time',
+            'ModuleKojiBuild': 'completion_time',
+            'ContainerKojiBuild': 'completion_time'
+        }
+
+        wait_times = []
+
+        for index in range(len_story - 1):
+            completion_time = getattr(results[index], completion_times[results[index].__label__])
+            start_time = datetime.strptime(results[index + 1].timeline_timestamp,
+                                           '%Y-%m-%dT%H:%M:%SZ')
+
+            # Remove timezone info so that both are offset naive and thus able to be subtracted
+            start_time = start_time.replace(tzinfo=None)
+            completion_time = completion_time.replace(tzinfo=None)
+
+            wait_time = start_time - completion_time
```

Comment:
should this be `completion_time - start_time` instead? Because `completion_time` should and will always be bigger right?

Reply:
@yashvardhannanavati I see what you're saying, but for here this is written this way because it is the start time of the next node compared with the completion of the current one, and we want the wait time to be the time from the current one finishing until the next one starts. Lmk if that doesn;t make sense still.

Reply:
makes sense. Thanks!